input-profile.ts

```diff
@@ -0,0 +1,272 @@
+    defaultController!: Object3D;
+
+    @property.enum(hands, 0)
+    handednessIndex: number = 0;
+
+    @property.string(
+        'https://cdn.jsdelivr.net/npm/@webxr-input-profiles/assets@latest/dist/profiles/'
+    )
+    path!: string;
+
+    @property.string()
+    customProfileFolder!: string;
+
+    @property.bool(false)
+    mapToDefaultController!: boolean;
+
+    @property.object()
+    trackedHand!: Object3D;
+
+    start() {
+        this._controllerModel = null;
+        this.toFilter = new Set(['vr-mode-active-mode-switch']);
+        this._defaultControllerComponents = this.getComponents(this.defaultController);
+        this._handedness = hands[this.handednessIndex];
+
+        this.engine.onXRSessionStart.add(() => {
+            console.warn('adding event listener for inputChange');
+            this.engine.xr?.session.addEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        });
+    }
+
+    onDeactivate() {
+        this.engine.onXRSessionStart.add(() => {
+            this.engine.xr?.session.removeEventListener(
+                'inputsourceschange',
+                this.onInputSourcesChange.bind(this)
+            );
+        });
+    }
+
+    setHandTrackingControllers(controllerObject: Object3D) {
+        const HandtrackingComponent = this.trackedHand.getComponent(HandTracking);
+        if (!HandtrackingComponent) return;
+        /**@ts-ignore**/
+        HandtrackingComponent.controllerToDeactivate = controllerObject;
+    }
+
+    getComponents(obj: Object3D) {
+        if (obj == null) return;
+        const components: Component[] = [];
+        const stack = [obj];
+        while (stack.length > 0) {
+            const currentObj = stack.pop();
+            const comps = currentObj
+                .getComponents()
+                .filter((c: Component) => !this.toFilter.has(c.type));
+            components.push(...comps);
+            const children = currentObj.children || [];
+            // Push children onto the stack in reverse order to maintain the correct order
+            for (let i = children.length - 1; i >= 0; --i) {
+                stack.push(children[i]);
+            }
+        }
+        return components;
+    }
+
+    setComponentsActive(active: boolean) {
+        const comps = this._defaultControllerComponents;
+        if (comps == undefined) return;
+        for (let i = 0; i < comps.length; ++i) {
+            comps[i].active = active;
+        }
+    }
+
+    onInputSourcesChange(event: XRInputSourceChangeEvent) {
+        if (this._modelLoaded && !this.mapToDefaultController) {
+            this.setComponentsActive(false);
+        }
+
+        event.added.forEach((xrInputSource: XRInputSource) => {
+            if (xrInputSource.hand != null) return;
+            if (this._handedness != xrInputSource.handedness) return;
+            var profile = this.path + xrInputSource.profiles[0];
```

Comment:
Use `const` instead

Reply:
We are updating profile in the next line if customProfileFolder is available , we cant reassign a const.

Reply:
Modified it this way 

```js
const profile =this.customProfileFolder !== ""
			? this.customProfileFolder
			: this.path + xrInputSource.profiles[0];
```